pyccel/codegen/printing/ccode.py

```diff
@@ -1134,35 +1181,51 @@ def formatted_args_to_printf(args_format, args, end):
             code += formatted_args_to_printf(args_format, args, end)
         return code
 
-    def find_in_dtype_registry(self, dtype):
+    def find_in_type_registry(self, dtype):
         """
-        Find the corresponding C dtype in the dtype_registry.
+        Find the corresponding C dtype in the dtype_registry or the STC container type.
 
-        Find the corresponding C dtype in the dtype_registry.
-        Raise PYCCEL_RESTRICTION_TODO if not found.
+        This function searches for the corresponding C data type in the `dtype_registry`.
+        If the provided dtype is a container (like `HomogeneousSetType` or
+        `HomogeneousListType`), it identifies the scalar type of the container and returns
+        the appropriate type for the `STC` container.
+        It raises a `PYCCEL_RESTRICTION_TODO` error if the dtype is not found in the registry.
```

Comment:
Sorry, but I'm not sure this name and docstring make sense. A "dtype" as per the docstring definition:
https://github.com/pyccel/pyccel/blob/20aab161e912483e9e8a85fca771e21b2d432bbb/pyccel/ast/basic.py#L543-L552
cannot be a container.


I suggest:

```python
    def get_c_type(self, dtype):
        """
        Find the corresponding C type of the PyccelType.

        For scalar types, this function searches for the corresponding C data type
        in the `dtype_registry`.  If the provided type is a container (like
        `HomogeneousSetType` or `HomogeneousListType`),  it recursively identifies
        the type of an element of the container and uses it to calculate the
        appropriate type for the `STC` container.
        A `PYCCEL_RESTRICTION_TODO` error is raised if the dtype is not found in the registry.
```

Reply:
updated 8b4fa11b2c6dce02e30c77cef4d60da0d58e9315